### PR TITLE
Remove ClientRequestId property from requests

### LINKED ISSUE #12109: Remove ClientRequestId from SearchClientOptions
Based on team discussion, remove `SearchClientOptions.ClientRequestId`.

### PR DESCRIPTION
**Pull request by heaths**
Fixes #12109

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 1]
```diff
@@ -188,7 +188,6 @@ public partial class SearchOptions : Azure.Search.Documents.SearchRequestOptions
     public partial class SearchRequestOptions
```
Comment (pakrym) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:188]: Empty type. Worth removing?

Comment (heaths) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:188]: Working on that now. @tg-msft and I were discussing reasons to keep it (none, really) or remove it entirely.

File: sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs [hunk 2]
```diff
@@ -689,8 +644,7 @@ public virtual SearchClient GetSearchClient(string indexName)
 
                     Response<ListIndexesResult> result = IndexesClient.List(
                         Constants.All,
-                        options?.ClientRequestId,
-                        cancellationToken);
+                            cancellationToken);
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs:647]: Nit - weird indentation.  Don't bother if there are no other changes needed.

File: sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs [hunk 3]
```diff
@@ -726,8 +678,7 @@ public virtual SearchClient GetSearchClient(string indexName)
 
                     Response<ListIndexesResult> result = await IndexesClient.ListAsync(
                         Constants.All,
-                        options?.ClientRequestId,
-                        cancellationToken)
+                            cancellationToken)
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs:681]: Nit - Same here.

File: sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs [hunk 4]
```diff
@@ -802,8 +748,7 @@ public virtual SearchClient GetSearchClient(string indexName)
 
                     Response<ListIndexesResult> result = await IndexesClient.ListAsync(
                         Constants.NameKey,
-                        options?.ClientRequestId,
-                        cancellationToken)
+                            cancellationToken)
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs:751]: Nit - Maybe the indentation thing was a weird find/replace?

**Review by tg-msft — APPROVED** [review]
Looks good - thanks for making this fix!

File: sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs (hunk 4 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs:751]: Yeah, I was using the regex search/replace. The whole line got selected, so not sure what happened here.

**Pull request merged**